src/l3-resources/atlas-bootstrap/index.ts

```diff
@@ -0,0 +1,129 @@
+import {
+  CfnOutput,
+  aws_iam as iam,
+  aws_secretsmanager as secretsmanager,
+  aws_cloudformation as cloudformation,
+  SecretValue,
+  Duration,
+} from "aws-cdk-lib";
+import { Construct } from "constructs";
+
+export const AtlasBasicResources: string[] = [
+  "Cluster",
+  "Project",
+  "DatabaseUser",
+  "ProjectIpAccessList",
+];
+
+export class MongoAtlasBootstrapProps {
+  /**
+   * The IAM role name for CloudFormation Extension Execution.
+   * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/registry-public.html
+   * @default auto generat the name.
+   */
+  readonly roleName!: string;
+  /**
+   * The secret profile name for MongoDB Atlas.
+   * @default generate a dummy secret.
+   * @see https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master#2-configure-your-profile
+   */
+  readonly secretProfile!: string;
+  /**
+   * List of strings representing mongoDB atlas types to activate.
+   * the exported AtlasBasicResources can be used for AtlasBasic resources
+   * if not provided non resources will be activated
+   * @example typesToActivate=["Cluster","Project"] this will actiate MongoDB::Atlas::Project && MongoDB::Atlas::Cluster
+   */
+  readonly typesToActivate?: string[];
+}
+
+/**
+ * Generate the CFN extension execution role.
+ * @see https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/registry-public.html
+ */
+export class MongoAtlasBootstrap extends Construct {
+  readonly role: iam.IRole;
+  constructor(scope: Construct, id: string, props?: MongoAtlasBootstrapProps) {
+    super(scope, id);
+
+    this.role = new iam.Role(this, "mongoDBAtlasCFNExecutionRole", {
+      maxSessionDuration: Duration.seconds(8400),
```

Comment:
What will happen if we leave `maxSessionDuration` undefined?

Reply:
This comes as a default property when you create an execution role with CFN, i just replicated a role exactly as they are created by the CFN tool